README.md

```diff
@@ -1,55 +1,156 @@
+It performs the following steps:
+1. Checks if `connectivityproxy` CRD is installed on the cluster, and exits with success if it is not the case
+2. Finds default `connectivityproxy CR`
+3. Exits with success if the `connectivityproxy CR` doesn't exist
+4. Exits with success if the `connectivityproxy CR` `migrator.kyma-project.io/backed-up="true"` annotation (backup was completed already)
+5. Creates `connectivity-proxy-backup` namespace if it doesn't exist
+6. Copies `connectivity-proxy`, and `connectivity-proxy-info` config maps into `connectivity-proxy-backup` namespace
+7. Adds `migrator.kyma-project.io/backed-up="true"` annotation on the CR.
 
-## Contributing
-<!--- mandatory section - do not change this! --->
+### Cleaner
 
-See the [Contributing Rules](CONTRIBUTING.md).
+1. Checks if `connectivityproxy` CRD is installed on the cluster, and exits with success if it is not the case
+2. Finds default `connectivityproxy CR`
+3. Exits with success if the `connectivityproxy CR` doesn't exist
+4. Exits with success if the `connectivityproxy CR` `migrator.kyma-project.io/cleaned="true` annotation (cleanup was completed already)
+5. Removes the legacy Connectivity Proxy 2.9.3 Kubernetes objects from the cluster (`deployments`, `statuefulsets`, `services`, `configmaps` etc.)
+6. Adds `migrator.kyma-project.io/cleaned="true"` annotation on the CR.
 
-## Code of Conduct
-<!--- mandatory section - do not change this! --->
 
-See the [Code of Conduct](CODE_OF_CONDUCT.md) document.
+## Migration process
+### Prerequisites
 
-## Licensing
-<!--- mandatory section - do not change this! --->
+  - Kyma Cluster with legacy Connectivity Proxy component installed 
+  - New Connectivity Proxy Kyma module ready to be enabled on Kyma cluster
+  - Module configured to include Connectivity Proxy init containers.
 
-See the [license](./LICENSE) file.
+### Implementation details
+
+The migration task will be automatically completed during startup of the `Connectivity Proxy Operator` pod, and implemented with init containers. \
+This Deployment and the default `ConnectivityProxy custom resource (CR)` will be installed on the Kyma cluster when new `Connectivity Proxy Kyma module` is enabled by the user.
+
+The Connectivity Proxy Operator pod runs Connectivity Proxy Migrator components as init containers in following order:
+
+1. **Configuration Migrator** - Extracts the user configuration from config maps and saves it into `ConnectivityProxy CR`.
+2. **Backup** - Backup of the user configuration into separate namespcace `connectivity-proxy-backup`.
+3. **Cleaner** - Deletes previous installation of connectivity proxy.
+
+**Note:**
+> - Correct RBACs object must be installed on the cluster to allow for execution of init containers.
+> - Init containers must be configured to communicate over the network without Istio sidecar. This can be done by using special **UID: 1337** \
```

Comment:
```suggestion
> - Init containers must be configured to communicate over the network without the Istio sidecar. This can be done by using special **UID: 1337** \
```